Author: Julien Cristau <jcristau@debian.org>
Description:
 Cherry-pick "[terminal] don't stomp over original tty lock settings". Prevents
 setting tty1 in 'echo' mode after boot (Closes: 595178).

diff -Naurp plymouth.orig/src/libply-splash-core/ply-terminal.c plymouth/src/libply-splash-core/ply-terminal.c
--- plymouth.orig/src/libply-splash-core/ply-terminal.c	2010-09-14 21:16:57.185256706 +0200
+++ plymouth/src/libply-splash-core/ply-terminal.c	2010-10-13 17:58:48.752970212 +0200
@@ -188,7 +188,8 @@ ply_terminal_set_unbuffered_input (ply_t
   if (tcsetattr (terminal->fd, TCSANOW, &term_attributes) != 0)
     return false;
 
-  if (ioctl (terminal->fd, TIOCGLCKTRMIOS, &locked_term_attributes) == 0)
+  if (!terminal->original_locked_term_attributes_saved &&
+      ioctl (terminal->fd, TIOCGLCKTRMIOS, &locked_term_attributes) == 0)
     {
       terminal->original_locked_term_attributes = locked_term_attributes;
       terminal->original_locked_term_attributes_saved = true;
